src/services/api.service.ts

```diff
@@ -0,0 +1,138 @@
+        });
+        if (this.localCustomUserAgent != null) {
+            headers.set('User-Agent', this.localCustomUserAgent);
+        }
+
+        const requestInit: RequestInit = {
+            cache: 'no-store',
+            credentials: this.localGetCredentials(),
+            method: method,
+        };
+
+        if (authed) {
+            const authHeader = await this.getActiveBearerToken();
+            headers.set('Authorization', 'Bearer ' + authHeader);
+        }
+        if (body != null) {
+            if (typeof body === 'string') {
+                requestInit.body = body;
+                headers.set('Content-Type', 'application/x-www-form-urlencoded; charset=utf-8');
+            } else if (typeof body === 'object') {
+                if (body instanceof FormData) {
+                    requestInit.body = body;
+                } else {
+                    headers.set('Content-Type', 'application/json; charset=utf-8');
+                    requestInit.body = JSON.stringify(body);
+                }
+            }
+        }
+        if (hasResponse) {
+            headers.set('Accept', 'application/json');
+        }
+
+        requestInit.headers = headers;
+        const response = await this.fetch(new Request(apiUrl + path, requestInit));
+
+        if (hasResponse && response.status === 200) {
+            const responseJson = await response.json();
+            return responseJson;
+        } else if (response.status !== 200) {
+            const error = await this.localHandleError(response, false, authed);
+            return Promise.reject(error);
+        }
+    }
+
+    private localGetCredentials(): RequestCredentials {
+        if (!this.localIsWebClient || this.localUsingBaseUrl) {
+            return 'include';
+        }
+        return undefined;
+    }
+
+    private async localHandleError(response: Response, tokenError: boolean, authed: boolean): Promise<ErrorResponse> {
+        if (authed && ((tokenError && response.status === 400) || response.status === 401 || response.status === 403)) {
+            await this.localLogoutCallback(true);
+            return null;
+        }
+
+        let responseJson: any = null;
+        if (this.localIsJsonResponse(response)) {
+            responseJson = await response.json();
+        } else if (this.localIsTextResponse(response)) {
+            responseJson = { Message: await response.text() };
+        }
+
+        return new ErrorResponse(responseJson, response.status, tokenError);
+    }
+
+    private localIsJsonResponse(response: Response): boolean {
+        const typeHeader = response.headers.get('content-type');
+        return typeHeader != null && typeHeader.indexOf('application/json') > -1;
+    }
+
+    private localIsTextResponse(response: Response): boolean {
+        const typeHeader = response.headers.get('content-type');
+        return typeHeader != null && typeHeader.indexOf('text') > -1;
+    }
+}

```

Comment:
why do we had to fork this service? What did we add?

Reply:
This is the joy of trying to inherit classes with private members 😓 

I added an override to `getCollections` (public) which depends on `send()` (private) which depends on `getCredentials()` (private) which depends on `handleError()` (private) which depends on both `IsJsonResponse()` (private) and `IsTextResponse()` (private)
So I had to reimplement all of those methods 😐